code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -132,3 +132,14 @@
 			cell.forceMove(get_turf(loc))
 			cell = null
 			update_icon()
+
+/obj/item/clothing/gloves/fingerless/rapid
+	name = "Gloves of the north star"
+	desc = "Just looking at these fills you with an urge to beat the shit out of people."
+
+/obj/item/clothing/gloves/fingerless/rapid/Touch(mob/living/target, proximity = TRUE)
+	var/mob/living/M = loc
+
+	if(M.a_intent == INTENT_HARM)
+		M.changeNext_move(CLICK_CD_RAPID)
```

Comment:
What handles resetting this value?

Reply:
Yes, but if someone punches somebody, then pulls out their esword, would they retain the fast attack speed?

Reply:
Oh, wait - I think I might have the vars that allow fast attack still on the brain - this is to set a single cooldown timer, yes?

Reply:
Correct.